#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure apr_ino_t stays the same regardless of
## DP: -D_FILE_OFFSET_BITS=64 or not
## DP: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=616323
@DPATCH@
Index: configure.in
===================================================================
--- a/configure.in
+++ b/configure.in
@@ -1726,9 +1726,13 @@
 # ino64_t as apr_off_t is off64_t, but this can't be done now without
 # breaking ABI.
 ino_t_value=ino_t
-if test "$ac_cv_sizeof_long" = "4"; then
-    APR_CHECK_TYPES_COMPATIBLE(ino_t, unsigned long, 
-                               ino_t_value="unsigned long")
+APR_CHECK_SIZEOF_EXTENDED(AC_INCLUDES_DEFAULT, ino_t, $ac_cv_sizeof_long)
+if test $ac_cv_sizeof_ino_t = 4; then
+    if test $ac_cv_sizeof_long = 4; then
+        ino_t_value="unsigned long"
+    else
+        ino_t_value="unsigned int"
+    fi
 fi
 AC_MSG_NOTICE([using $ino_t_value for ino_t])
 
